litestar/plugins.py

```diff
@@ -174,7 +174,7 @@ class PluginRegistry:
 
     def __init__(self, plugins: list[PluginProtocol]) -> None:
         self._plugins_by_type = {type(p): p for p in plugins}
-        self._plugins = frozenset(plugins)
+        self._plugins = set(plugins)
```

Comment:
Why this change?
The reason this was a frozenset is because we don't allow registering plugins dynamcially. If we want to, we need to take care of a bunch of stuff, e.g. regenerating the `openapi`, `init` and erialization` plugin tuples.

Reply:
Sorry, this was not meant to be included here. This was part of my socketify debugging.

Reply:
```suggestion
        self._plugins = frozenset(plugins)
```